pkg/kubernetes/kubernetes.go

```diff
@@ -455,26 +456,29 @@ func getCRDFromManifest(ctx context.Context, fileName string) (*apiextensionsv1b
 
 // GetDynamicInformer returns informer for specified CRD group, version and name
 // return error if failure observed
-func GetDynamicInformer(ctx context.Context, crdGroup, crdVersion, crdName string) (informers.GenericInformer, error) {
+func GetDynamicInformer(ctx context.Context, crdGroup, crdVersion, crdName, namespace string) (informers.GenericInformer, error) {
 	log := logger.GetLogger(ctx)
-	dynamicInformerInitLock.Lock()
-	defer dynamicInformerInitLock.Unlock()
 
-	if !dynamicInformerInitialized {
-		// Get a config to talk to the apiserver
-		cfg, err := GetKubeConfig(ctx)
-		if err != nil {
-			log.Errorf("failed to get Kubernetes config. Err: %+v", err)
-			return nil, err
-		}
-		// Grab a dynamic interface to create informers from
-		dc, err := dynamic.NewForConfig(cfg)
-		if err != nil {
-			log.Errorf("could not generate dynamic client for config. err :%v", err)
-			return nil, err
+	if atomic.LoadUint32(&dynamicInformerInitialized) == 0 {
```

Comment:
Can you explain how `atomic.LoadUint32(&dynamicInformerInitialized)` help here?

Reply:
I just changed this code to implement double locking mechanism. We have used this approach to create singleton objects in other parts of code as well and it is foolproof.

Reply:
Removed the double locking based on our discussion.